[16기 유재현] React Westagram Login,Main Page Layout 및 필수 구현 사항 완료

JaehyunYoo opened the pull request:
# 👍 React Westagram Team - 2 Project

✅ &nbsp; Login Page : React Login 페이지 필수 구현 완료 

✅ &nbsp; Main Page : React Main 페이지 필수 구현 완료 

`추가 구현사항 및 구현하고 싶은 기능 구현중`

## 💡 &nbsp;수정 사항 간략한 한줄 요약
- ✅ &nbsp; 로그인 페이지 필수 구현사항 완료 
- ✅ &nbsp;메인페이지 컴포넌트 단위로 구성 및 Sass 정리 추가 기능 구현중
- ✅ &nbsp; Self review 진행 완료
- ☑️ &nbsp;Peer Review 정리 및 분리 (준비중)

- `기타 수정 및 가이드라인 맞추어 작업중`


		 
## 🔹 &nbsp; 수정_ 사항들 자세한 내용
### 로그인페이지
- [x] id , password 4자리 이상 입력시 버튼활성화 기능 구현완료
- [x] id @ 미포함시 alert 메세지 출력
- [x] 모든 조건 충족시 Main 페이지로 이동 구현 완료
### 메인페이지
- [x] 메인페이지 안에 있는 요소들 전체를 컴포넌트단위로 작업 완료 
- [x] 댓글 입 력시 this.setState를 이용하여 업데이트 및 Mock Data를 활용 하여 Data.json 형식 Fetch 함수를 사용하여 구현완료
- [x] props 를 이용하여 함수를 전달 후 메인컴포넌트에서 filter를 활용하여 id값이 일치 하지 않는 요소들을 this.setState해서 업데이트를 하여 삭제 구현 완료 
- [x] 좋아요 버튼 구현완료 props 로 전달 받은 데이터를 state에서 가져온 후 this.setState를 이용하여 업데이트 하였습니다. 
- [x] Follow Component 를 불필요한 요소가 반복되는게 보여 다시 Component 분리 하여 작업 완료 (Mock Data 활용)
- [ ] Right Feed Story Component 로 구성하여 재사용 가능 할 수 있도록 구현중
## 기타 질문 및 특이 사항
- ~~id를 사용하여 해당 인풋에 접근하였는데 조금 공부한바로는 React Ref라는  것을 찾았습니다. 하지만
어떤상황에 써야하는지 아직 익숙치 않습니다. 구글링을 찾아보긴 했지만 아 이거구나 하는 이해는 아직 하지 못했습니다.~~
- ~~this.props 를 아직 완전이 이해하지 못한 부분이 있습니다 . props 로 함수 전달시 인자를 무조건 받아햐 하는 것인지 알고 싶습니다 아닐 경우는 어떤경우 인지가 알고싶습니다.~~

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

JaehyunYoo committed:
Add:Login 페이지 필수 구현사항 완료

JaehyunYoo committed:
Add:MainPage 및 Component Navbar , Feed, RightFeed Followers 추가 구현 완료

JaehyunYoo committed:
Add:댓글 기능 추가 완료

JaehyunYoo committed:
Modify: Mock Data 활용 구현중

JaehyunYoo committed:
Modify:Follow Component 생성 및 Mock Data로 수정 구현 완료

JaehyunYoo committed:
"Modify: Right Feed, Follow,Right Component 분리 수정 완료"
"Add:Fetch Mock Data 구현 완료 "
"Add:Delete Button Click 구현 완료"

JaehyunYoo committed:
Add: 좋아요 버튼 구현 props를 이용하여 함수를 전달 구현완료

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Feed.js:
```diff
@@ -0,0 +1,158 @@
+    }
+    componentDidMount() {
+        fetch('http://localhost:3000/data/Comment.json', {
+            method: 'GET',
```
fetch함수의 get method는 method의 default 값입니다. 따라서 생략이 가능합니다.

JaehyunYoo replied:
넵 아직 커밋을 안해서 업데이트 하여 반영하겟습니다.

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Feed.js:
```diff
@@ -0,0 +1,158 @@
+            commentValue: '',
+            isLiked: false,
+        };
+    }
+    componentDidMount() {
```
메서드 사이 간격 한 칸 띄워주세요!

JaehyunYoo replied:
넵 알겟습니다 수정 후 업데이트 하겠습니다.

dipsiiiiiiiiii commented on src/Routes.js:
```diff
@@ -1,48 +1,42 @@
-
-
+// import LoginKim from "./pages/byungjinkim/Login/Login";
+// import MainKim from "./pages/byungjinkim/Main/Main";
```
다른 팀원 분들 컴포넌트를 주석처리 하시면 나중에 main 브랜치에서 머지가 될 때 컨플릭트가 납니다. 컴포넌트 안의 내용이 없어서 에러가 뜬다면 기본 컴포넌트 구조를 넣어주세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Main/Main.js:
```diff
@@ -0,0 +1,34 @@
+class Main extends Component {
+    render() {
+        return (
+            <div className='main'>
```
가장 바깥 태그의 클래스명은 컴포넌트 명과 동일하게 PascalCase로 작성해 주세요! wecode 컨벤션입니다!

그리고 들여쓰기 칸이 4칸일 경우 코드가 가로로 길어져 가독성에 좋지 않습니다.
vscode 설정에서 tab칸을 2칸으로 줄여 가독성을 높여주세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Login/Login.scss:
```diff
@@ -0,0 +1,82 @@
+@import url(https://fonts.google.com/specimen/Lobster?preview.text=&preview.text_type=custom);
+
+* {
+    box-sizing: border-box;
+}
```
초기화 속성은 reset에 넣어 주세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Login/Login.scss:
```diff
@@ -0,0 +1,82 @@
+    justify-content: center;
+    overflow: hidden;
+    text-align: center;
+    background-color: #fafafa;
+    .login__wrap {
```
속성과 선택자 사이 간격 띄워 주세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Login/Login.js:
```diff
@@ -0,0 +1,104 @@
+        // }
+    };
+    handleButton = () => {
+        fetch(`${API}`, {
```
template literals이 필요하지 않은 것 같네요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/RightFeed/RightFeed.js:
```diff
@@ -0,0 +1,159 @@
+                </div>
+
+                <ul className='navigation__footer'>
+                    <li>
+                        <Link to=''>소개</Link>
+                    </li>
+                    <li>
+                        <Link to=''>도움말</Link>
+                    </li>
+                    <li>
+                        <Link to=''>홍보센터</Link>
+                    </li>
+                    <li>
+                        <Link to=''>API</Link>
+                    </li>
+                    <li>
+                        <Link to=''>채용정보</Link>
+                    </li>
+                    <li>
+                        <Link to=''>개인정보처리방침</Link>
+                    </li>
+                    <li>
+                        <Link to=''>약관</Link>
+                    </li>
+                    <li>
+                        <Link to=''>위치</Link>
+                    </li>
+                    <li>
+                        <Link to=''>인기계정</Link>
+                    </li>
+                    <li>
+                        <Link to=''>해치태그</Link>
+                    </li>
+                    <li>
+                        <span>언어</span>
+                    </li>
```
이렇게 반복되는 태그의 묶음이 있다면 하나의 컴포넌트로 분리할 수 있습니다. 그리고 map 메서드를 활용해서 구현해 보세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/RightFeed/RightFeed.js:
```diff
@@ -0,0 +1,159 @@
+                        <Link to=''>모두 보기</Link>
+                    </header>
+                    <ul className='list__container'>
+                        <li>
+                            <div className='recomed__row'>
+                                <span className='round'>
+                                    <img
+                                        src='images/recom__img1.jpg'
+                                        alt='recommend__user'
+                                    />
+                                </span>
+                                <div>
+                                    <span className='d_block'>yoona_lim</span>
+                                    <span className='time'>
+                                        회원님을 위한 추천
+                                    </span>
+                                </div>
+                            </div>
+                            <button type='type='>팔로우</button>
+                        </li>
```
이렇게 반복되는 태그의 묶음이 있다면 하나의 컴포넌트로 분리할 수 있습니다. 그리고 map 메서드를 활용해서 구현해 보세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Follows/Follows.js:
```diff
@@ -0,0 +1,31 @@
+    }
+    componentDidMount() {
+        fetch('http://localhost:3000/data/FollowData.json', {
+            method: 'GET',
```
get 메서드가 기본 값이므로 get 요청 시 두 번째 인자는 생략 가능

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Follows/Follow.js:
```diff
@@ -0,0 +1,22 @@
+                {this.props.productList.map((item) => {
+                    return (
+                        <figure key={item.id}>
+                            <span className='recom__img'>
```
클래스명도 camelCase로 해주세요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Navbar.js:
```diff
@@ -0,0 +1,87 @@
+                        </div>
+                    </form>
+                    <ul className="navMenu__right">
+                        <li className="navMenu">
+                            <Link to="">{navbarSvg[0]}</Link>
+                        </li>
+                        <li className="navMenu">
+                            <Link to="">{navbarSvg[1]}</Link>
+                        </li>
+                        <li className="navMenu">
+                            <Link to="">{navbarSvg[2]}</Link>
+                        </li>
+                        <li className="navMenu">
+                            <Link to="">{navbarSvg[3]}</Link>
+                        </li>
```
map 메서드 활용할 수 있겠네요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Feed.js:
```diff
@@ -0,0 +1,158 @@
+                </figure>
+                <div className='feed__menu p_content'>
+                    <ul>
+                        <li>{feedSvg[0]}</li>
+                        <li>
+                            <Link to='/'>{feedSvg[1]}</Link>
+                        </li>
+                        <li>
+                            <Link to='/'>{feedSvg[2]}</Link>
+                        </li>
```
map 메서드 활용할 수 있겠네요!

dipsiiiiiiiiii commented on src/Pages/yoojaehyun/Components/Feed.js:
```diff
@@ -0,0 +1,158 @@
+                        <strong>d_a__m_i</strong>
+                    </span>
+                    <p>
+                        오늘 만나서 너무 반가웠어요! 16기 위코더 분들 항상
+                        응원합니다. 🥰
```
재현님을 못뵈서 아쉽네요 ㅠㅠ

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다!
리뷰 반영해 주세요!

JaehyunYoo committed:
1. 초기화 속성 reset파일 추가
2. 클래스명과 컴포넌트 명과 동일한 이름 수정
3. Route 팀원들간의컴포넌트 Route 설정 주석 제거
4. 메서드 사이 간격 수정
5. fetch 함수의 get method 제거
